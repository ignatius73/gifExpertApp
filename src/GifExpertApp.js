import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = []}) => {

    const [categories, setcategories] = useState(defaultCategories);


   /* const handleAdd = () => {
        const nuevaCategoria = 'Doraemon';
        setcategories([...categories, nuevaCategoria]);
    }*/

    return ( 
        <div>
            <h2> GifExpertApp</h2>
            <AddCategory setCategories = { setcategories }/>
        <hr / >

        <ol > {
            /*categories.map( cat => {
                return <li key= { cat }> { cat } < /li>;
            })*/
            categories.map( cat => (
                    <GifGrid 
                        key= {cat}
                        category= { cat }
                        /> )  
            )} 
            </ol> </div>
    )
}