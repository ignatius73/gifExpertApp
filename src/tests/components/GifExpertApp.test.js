import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../../GifExpertApp';

describe('Unitary tests over GifExpertApp', () => {
    

    test('should match with the Snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show a list of Categories', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories = { categories }/>);
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
    
})

