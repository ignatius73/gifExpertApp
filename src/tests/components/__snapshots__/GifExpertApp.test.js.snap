// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitary tests over GifExpertApp should match with the Snapshot 1`] = `
<div>
  <h2>
     GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
     
  </ol>
   
</div>
`;
